data/src/message.rs

```diff
@@ -5,14 +5,32 @@ use serde::{Deserialize, Serialize};
 use crate::{time, User};
 
 pub type Raw = irc::proto::Message;
+pub type Channel = String;
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub enum Source {
     Server,
-    Channel(String, User),
+    Channel(Channel, Kind),
     Query(User),
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum Kind {
+    /// `Kind::User(_)` is coming from another client.
+    User(User),
+    /// `Kind::Server()` is from the server, targeting a channel.
+    Server,
+}
+
+impl Kind {
+    pub fn user(&self) -> Option<&User> {
+        match self {
+            Kind::User(user) => Some(user),
+            Kind::Server => None,
+        }
+    }
+}
+
```

Comment:
`Kind` is a bit generic since it only applies to `Channel`. Maybe we call it `ChannelSender`?